### PR TITLE
React app setup

### PR DESCRIPTION
**Pull request by ritchieritch42**
Copied React-cors-spa from AWS documentation

Removed async and it's package.json definition

Removed additional SAM template

Removed SimpleApi as our RestApi resource is robust and meets all project needs

Cloudfront Distribution modified to match for an origin path of "um/"

### CODE REVIEW

File: template.yml [hunk 1]
```diff
@@ -96,6 +96,127 @@ Resources:
           /category/delete/{category_id}:
             delete: *standardImplementation
 
+  # The AWS S3 bucket into which our SPA build files will be deployed
+  S3Bucket:
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName: !Sub "react-cors-spa-${RestApi}"
```
Comment (Andrewdt97) [inline template.yml]: Why is the api here?

File: template.yml [hunk 2]
```diff
@@ -96,6 +96,127 @@ Resources:
           /category/delete/{category_id}:
             delete: *standardImplementation
 
+  # The AWS S3 bucket into which our SPA build files will be deployed
+  S3Bucket:
```
Comment (Andrewdt97) [inline template.yml]: Can we name this something more specific?

File: template.yml [hunk 3]
```diff
@@ -106,3 +227,9 @@ Outputs:
   BackendLambdaFunctionArn:
     Description: "ARN of the Backend Lambda"
     Value: !GetAtt BackendLambda.Arn
+  APIEndpoint:
+    Value: !Sub "https://${RestApi}.execute-api.${AWS::Region}.amazonaws.com/v1/um"
```
Comment (Andrewdt97) [inline template.yml]: You can remove `v1/um`

File: template.yml [hunk 4]
```diff
@@ -96,6 +96,127 @@ Resources:
...
+                AWS:SourceArn: !Sub arn:aws:cloudfront::${AWS::AccountId}:distribution/${CFDistribution}
+            Action: "s3:GetObject*"
+      Bucket: !Ref S3Bucket
+
+  # The Amazon S3 bucket into which access logs from S3 (for the application) and CloudFront will be put
+  LoggingBucket:
+    #checkov:skip=CKV_AWS_18: "This bucket is private and only for storing logs"
+    Type: "AWS::S3::Bucket"
+    Properties:
+      BucketName: !Sub "react-cors-spa-${RestApi}-logs"
```
Comment (Andrewdt97) [inline template.yml]: And here?

**Comment by Andrewdt97** [comment]
A couple specific questions but we still need two things. I think I forgot the first in the story my bad.

1) We need to deploy this stack to AWS to create the bucket and the rest of the resources. I would do a dry run first because CF is notoriously bad when it tries to deploy something invalid.
2) We still need a script to being to build the web app and upload it into the bucket.
3) Make sure you can get to the website via the cloud front url. Once you deploy the stack, you should be able to go into cloud front and find this.

File: template.yml (hunk 1 above)
Comment (ritchieritch42) [reply template.yml]: to use as the name of the bucket

File: template.yml (hunk 2 above)
Comment (ritchieritch42) [reply template.yml]: Yes! How does IM-React-App sound?

File: template.yml (hunk 3 above)
Comment (ritchieritch42) [reply template.yml]: done, should I also remove on line 189 for me the pathpattern: "v1/im" ?

File: template.yml (hunk 4 above)
Comment (ritchieritch42) [reply template.yml]: name of the bucket again, but can change to whatever!

**Comment by ritchieritch42** [comment]
> A couple specific questions but we still need two things. I think I forgot the first in the story my bad.
> 
>     1. We need to deploy this stack to AWS to create the bucket and the rest of the resources. I would do a dry run first because CF is notoriously bad when it tries to deploy something invalid.
> 
>     2. We still need a script to being to build the web app and upload it into the bucket.
> 
>     3. Make sure you can get to the website via the cloud front url. Once you deploy the stack, you should be able to go into cloud front and find this.

File: template.yml (hunk 1 above)
Comment (Andrewdt97) [reply template.yml]: I wouldn't. The bucket has little to do with the api. I would include the name of the project maybe but you don't need the api id.

**Comment by ritchieritch42** [comment]
@Andrewdt97 I was going to write the script as a separate pull request, is that alright?

**Comment by ritchieritch42** [comment]
Also I accidentally closed the request woops

File: template.yml (hunk 3 above)
Comment (Andrewdt97) [reply template.yml]: Oh yes. Get rid of that whole block.

**Comment by Andrewdt97** [comment]
> @Andrewdt97 I was going to write the script as a separate pull request, is that alright?

That's fine but can we at least get the dry run of the CF deployment so we know this is all good according to AWS?

**Comment by ritchieritch42** [comment]
Yes I'll dry deploy this morning

**Pull request merged**